forest/daemon/src/daemon.rs

```diff
@@ -377,8 +377,20 @@ pub(super) async fn start(config: Config, detached: bool) -> Db {
 
     services.spawn(p2p_service.run());
 
-    // Block until ctrl-c is hit
-    ctrlc_oneshot.await.unwrap();
+    loop {
+        select! {
+            option = services.join_next().fuse() => {
+                if let Some(Ok(Err(error_message))) = option {
+                    let msg = format!("services failure: {}", error_message);
+                    error!("{}", msg);
+                    break
+                }
+            },
+            complete => select! {
+                _ = ctrlc_oneshot => break
+            },
+        }
+    }
```

Comment:
I'm 90% sure that this code does not work.

Reply:
Changed this logic